### PR TITLE
EF implementation

### PR DESCRIPTION
**Pull request by pskramtai**
--Mapped base classes in DBContext.
--Added some methods to get/save data in DBAgent (not complete, don't know how to get DbContextOptions from DI container).

### CODE REVIEW

File: WebService/WebService/Data/DbAgent.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                using (var dbContext = new DBContext(options))
+                {
+                    return dbContext.Bars.ToList();
+                }
+            }
+            catch (Exception e)
+            {
+                
+                throw;
+            }
```
Comment (ElektrosStulpas) [inline WebService/WebService/Data/DbAgent.cs:27]: empty whitespace :D where's the convention that we agreed on, Lebowski???
But for real, we need to start thinking about logging exceptions and not just throwing them left and right

File: WebService/WebService/Data/DbAgent.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using WebService.Base_Classes;
+
+namespace WebService.Data
+{
+    public class DbAgent
+    {
+        //private string connectionString = "Server=tcp:comparison-engine.database.windows.net,1433;Initial Catalog=Comparison Engine;Persist Security Info=False;User ID=PSIadmin;Password=CompEngine123Ė;MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;";
```
Comment (ElektrosStulpas) [inline WebService/WebService/Data/DbAgent.cs:13]: I think this has to go to appsettings.json, no?

File: WebService/WebService/Data/DbAgent.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                    dbContext.SaveChanges();
+                }
+            }
+            catch (Exception)
+            {
+
+                throw;
+            }
+        }
+    }
```
Comment (ElektrosStulpas) [inline WebService/WebService/Data/DbAgent.cs:112]: The save functions only add a single object to db, right? Can we edit an object in db? What about remove?

File: WebService/WebService/Data/DbAgent.cs [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                {
+                    return dbContext.SpecificPrices.ToList();
+                }
+            }
+            catch (Exception)
+            {
+
+                throw;
+            }
+        }
```
Comment (ElektrosStulpas) [inline WebService/WebService/Data/DbAgent.cs:60]: All the get methods return lists of objects, will we store them in the memory? What about methods that get specific list of drinks when you input bar and specific list of bars when you input drink (should they be in another class)?

File: WebService/WebService/Base Classes/Bar.cs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        public void RemoveDrink(int drinkID)
+        {
+            availableDrinks.Remove(drinkID);
+        }
+
+        //Changes the price of a drink at this bar
+        public void EditDrinkPrice(int drinkID, float newPrice)
+        {
+            availableDrinks[drinkID] = newPrice;
+        }*/
```
Comment (ElektrosStulpas) [inline WebService/WebService/Base Classes/Bar.cs:62]: If we keep BarOld, there's no reason to keep comments from there

File: WebService/WebService/Base Classes/Drink.cs [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        public void EditPrice(int barID, float newPrice)
+        {
+            drinkLocations[barID] = newPrice;
+        }
+
+        //Removes the bar from the dictionary of available bars
+        public void RemoveAvailableBar(int barID)
+        {
+            drinkLocations.Remove(barID);
+        }*/
```
Comment (ElektrosStulpas) [inline WebService/WebService/Base Classes/Drink.cs:59]: If we keep DrinkOld, there's no reason to keep comments from there

**Review by ElektrosStulpas — CHANGES_REQUESTED** [review]
I don't understand why it added a bunch of stuff that was already in the project, you sure you were developing on the newest master?
Also, at least deleting the comments that I mentioned would be nice, the extra functionality comments are more for consideration, but we need to discuss that, or if they make sense, feel free to implement them.

File: WebService/WebService/Data/DbAgent.cs (hunk 1 above)
Comment (pskramtai) [reply WebService/WebService/Data/DbAgent.cs:27]: Left space for exception handling for later.

File: WebService/WebService/Data/DbAgent.cs (hunk 3 above)
Comment (pskramtai) [reply WebService/WebService/Data/DbAgent.cs:112]: These are just some of the main functions for beginning. We can easily add any function we want later.

**Pull request merged**